cli/tools/fmt.rs

```diff
@@ -521,6 +535,138 @@ pub fn format_html(
   })
 }
 
+/// A function for formatting embedded code blocks in JavaScript and TypeScript.
+fn create_external_formatter_for_typescript(
+  unstable_options: &UnstableFmtOptions,
+) -> impl Fn(
+  MediaType,
+  String,
+  &dprint_plugin_typescript::configuration::Configuration,
```

Comment:
Question I was just thinking about and didn't look into enough during the first review. I think markup_fmt exposes a `fileIndentLevel` that allows figuring out what the starting indentation level should be when formatting: https://github.com/denoland/deno/pull/27164/files#diff-922e01000758964a3d0058e0ce425faeef65a5706e990516329b9ba62db5e369R445

So for example, it might say the current indent level is level 5 and so any file being formatted will format with an initial indent level of 5 rather than 0.

Do we need to use that here or is that currently handled internally within dprint-plugin-typescript? In the future we should probably do something similar.

Reply:
Top-level indentation is handled by dprint-plugin-typescript. (Currently it supposes the external formatter formats the text starting with indentation level 0, then plugin-typescript adds the appropriate indent to them using the print item sequence of `Signal::NewLine` `Signal::StartIndent` ... embedded contents ... `Signal::FinishIndent`

ref https://github.com/dprint/dprint-plugin-typescript/blob/936cf945ad9f0e69bb31cffe1c85b1cf991fe56e/src/generation/generate.rs#L3041-L3066

So I don't think we need to use `fileIndentLevel` of markup_fmt.